addons/purchase/tests/test_purchase_invoice.py

```diff
@@ -704,3 +704,62 @@ def test_onchange_partner_no_currency(self):
 
         self.assertEqual(bill.currency_id, self.currency_data['currency'], "The currency of the Bill should be the one of the context")
         self.assertEqual(bill.invoice_line_ids.currency_id, self.currency_data['currency'], "The currency of the Bill lines should be the same as the currency of the Bill")
+
+    def test_payment_reference_autocomplete_invoice(self):
+        """
+        Test that the payment_reference field is not replaced when selected a purchase order
+        We test the flow for 8 use cases:
+        - Purchase order with partner ref:
+            - Bill with ref:
+                - Bill with payment_reference -> should not be replaced
+                - Bill without payment_reference -> should be the po.partner_ref
+            - Bill without ref:
+                - Bill with payment_reference -> should not be replaced
+                - Bill without payment_reference -> should be the po.partner_ref
+        - Purchase order without partner ref:
+            - Bill with ref
+                - Bill with payment_reference -> should not be replaced
+                - Bill without payment_reference -> should be the bill ref
+            - Bill with ref
+                - Bill with payment_reference -> should not be replaced
+                - Bill without payment_reference -> should be empty
+        """
+        purchase_order_w_ref, purchase_order_wo_ref = self.env['purchase.order'].with_context(tracking_disable=True).create([
+            {
+                'partner_id': self.partner_a.id,
+                'partner_ref': 'PO-001' if i == 0 else False,
+                'order_line': [
+                    (0, 0, {
```

Comment:
`Command.create`

Reply:
@guva-odoo Have you missed that one?

Reply:
@guva-odoo :point_up: 
```suggestion
                    Command.create({
```

Reply:
Oups, sorry! Done